test/mpi/test_alltoallv.cc

```diff
@@ -140,6 +156,20 @@ TestAlltoallv::TestAlltoallv(ucc_test_team_t &_team, TestCaseParams &params) :
         args.src.info_v.displacements = (ucc_aint_t*)sdispls;
         args.dst.info_v.displacements = (ucc_aint_t*)rdispls;
     }
+    if (is_onesided) {
+        if (TEST_FLAG_VSIZE_64BIT == displ_bits) {
+            uint64_t *ldisp = (uint64_t *)calloc(nprocs, sizeof(uint64_t));
+            MPI_Alltoall(args.dst.info_v.displacements, 1, MPI_LONG, ldisp, 1,
```

Comment:
Can you explain why we need a communication here? Does it mean that any user must perform such a communication before posting the collective ?

Reply:
The communication is necessary because the onesided algorithm requires different semantics for displacements compared to the twosided algorithm. In particular, the destination displacements need to reflect the remote PE's displacement in the destination buffer rather than the local displacement.